packages/langium/test/utils/cst-utils.test.ts

```diff
@@ -0,0 +1,28 @@
+/******************************************************************************
+ * Copyright 2022 TypeFox GmbH
+ * This program and the accompanying materials are made available under the
+ * terms of the MIT License, which is available in the project root.
+ ******************************************************************************/
+import { createLangiumGrammarServices, EmptyFileSystem, Grammar } from '../../src';
+import { parseHelper } from '../../src/test';
+import * as c from '../../src/utils/cst-util';
+
+const services = createLangiumGrammarServices(EmptyFileSystem);
+const parser = parseHelper<Grammar>(services.grammar);
+
+describe('CST Utils', () => {
+
+    test('Find Leaf Node at Offset', async () => {
+        const text = `
+        grammar test
+        Main: value=AB;
+        terminal fragment Frag: 'B';
+        terminal AB: 'A' Frag;
+        `;
+
+        const grammar = await parser(text);
+        const offset = grammar.parseResult.value.$document?.textDocument.getText().indexOf('AB');
+        const leafnode = c.findLeafNodeAtOffset(grammar.parseResult.value.$cstNode!, offset!);
```

Comment:
```suggestion
        const leafnode = c.findLeafNodeAtOffset(grammar.$cstNode!, offset!);
```

Reply:
From what I could tell, `$cstNode` is only available on the `parseResult`. Am I missing something?

Reply:
Yeah, sorry got confused, since I expected `grammar` to be an AST element, not a document. It's all good.